tests/foreman/cli/test_errata.py

```diff
@@ -714,757 +700,768 @@ def test_host_errata_search_commands(self):
+    :return: a list of errata info dict for each errata id in erratum_ids
+    :type erratum_ids: list[str]
+    :type sort_by: str
+    :type sort_reversed: bool
+    :rtype: list[dict]
+    """
+    if len(erratum_ids) > ERRATUM_MAX_IDS_INFO:
+        raise Exception('Erratum ids length exceeded')
+    erratum_info_list = []
+    for errata_id in erratum_ids:
+        erratum_info_list.append(Erratum.info(options={'id': errata_id}, output_format='json'))
+    sorted_erratum_info_list = sorted(
+        erratum_info_list, key=itemgetter(sort_by), reverse=sort_reversed
+    )
+    return sorted_erratum_info_list
+
+
+@pytest.mark.tier3
+@pytest.mark.upgrade
+def test_positive_list_sort_by_issued_date(module_org):
+    """Sort errata by Issued date
+
+    :id: d838a969-d70a-43ae-9805-2e94dd985d6b
+
+    :Setup: Errata synced on satellite server.
+
+    :Steps:
+
+        1. erratum list --order 'issued ASC'
+        2. erratum list --order 'issued DESC'
+
+    :expectedresults: Errata is sorted by Issued date.
+    """
+    sort_data = [('issued', 'ASC'), ('issued', 'DESC')]
+    for sort_field, sort_order in sort_data:
+        sort_text = f'{sort_field} {sort_order}'
+        sort_reversed = True if sort_order == 'DESC' else False
+        erratum_list = Erratum.list({'order': sort_text, 'per-page': ERRATUM_MAX_IDS_INFO})
+        # note: the erratum_list, contain a list of restraint info
+        # (id, errata-id, type, title) about each errata
+        assert len(erratum_list) > 0
+        # build a list of erratum id received from Erratum.list
+        erratum_ids = [errata['id'] for errata in erratum_list]
+        # build a list of errata-id field value in the same order as
+        # received from Erratum.list
+        errata_ids = [errata['errata-id'] for errata in erratum_list]
+        # build a sorted more detailed erratum info list, that also
+        # contain the sort field
+        sorted_errata_info_list = _get_sorted_erratum_ids_info(
+            erratum_ids, sort_by=sort_field, sort_reversed=sort_reversed
+        )
+        # build a list of sort field (issued/updated) values in the
+        # same order as received from the detailed sorted erratum info
+        # list
+        sort_field_values = [errata[sort_field] for errata in sorted_errata_info_list]
+        # ensure that the sort field (issued/updated) values are sorted
+        # as needed
+        assert sort_field_values == sorted(sort_field_values, reverse=sort_reversed)
+        # build a list of errata-id field value in the same order as
+        # received from the detailed sorted errata info list
+        sorted_errata_ids = [errata['errata-id'] for errata in sorted_errata_info_list]
+        # ensure that the errata ids received by Erratum.list is sorted
+        # as needed
+        assert errata_ids == sorted_errata_ids
+
+
+@pytest.mark.tier3
+def test_positive_list_filter_by_org_id_and_sort_by_updated_date(module_org, rh_repo, custom_repo):
```

Comment:
There is a large amount of repeated code throughout these functions, with the same statements being executed in several test cases with only small differences. The test case names here are unwieldy because of their length.

It looks like the org field could be parametrized against a single test function, iterating over `id, name, label` for the org field, and `updated, issued` for the sorting.  Even further, org/product could be parametrized.

This will build a test function that implements a functionality to test, with a matrix of parameters that apply to that functionality.  The current test structure forces a high maintenance cost for any changes or fixes for the tests, and seemingly little value when it comes to test complexity.

Reply:
Hi

I am sorry I did not have time to refactor this test, converting it to pytest is a big job in itself